testplan/common/utils/match.py

```diff
@@ -2,16 +2,18 @@
 Module of utility types and functions that perform matching.
 """
 import os
-import time
 import re
+import time
+from typing import List, Match, Optional, Pattern, Tuple, Union
 
-from . import timing
-from . import logger
+from . import logger, timing
 
 LOG_MATCHER_INTERVAL = 0.25
 
 
-def match_regexps_in_file(logpath, log_extracts):
+def match_regexps_in_file(
+    logpath: os.PathLike, log_extracts: List[Pattern]
+) -> Tuple[bool, dict, List[Pattern]]:
```

Comment:
Can we be more specific what sort of `dict` is coming back? If types are complex just go with some umbrella, but we like to do `Dict[..., ...]`.

Reply:
Sure thing, it actually contains extracted values with name of key and type of value specified by regular expression.

Reply:
oh actually it's just string, will fix